mongodbatlas/resource_mongodbatlas_online_archive.go

```diff
@@ -0,0 +1,414 @@
+			for _, partition := range list {
+				item := partition.(map[string]interface{})
+				localOrder := item["order"].(int)
+				localOrderFloat := float64(localOrder)
+				partitionList = append(partitionList,
+					&matlas.PartitionFields{
+						FieldName: item["field_name"].(string),
+						Order:     pointy.Float64(localOrderFloat),
+					},
+				)
+			}
+
+			requestInput.PartitionFields = partitionList
+		}
+	}
+
+	return requestInput
+}
+
+func resourceMongoDBAtlasOnlineArchiveUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*matlas.Client)
+
+	ids := decodeStateID(d.Id())
+
+	atlasID := ids["atlas_id"]
+	projectID := ids["project_id"]
+	clusterName := ids["cluster_name"]
+
+	// if the criteria or the paused is enable then perform an update
+	paused := d.HasChange("paused")
+	criteria := d.HasChange("criteria")
+
+	// nothing to do, let's go
+	if !paused && !criteria {
+		return nil
+	}
+
+	request := matlas.OnlineArchive{}
+
+	// reading current value
+	if paused {
+		request.Paused = pointy.Bool(d.Get("paused").(bool))
+	}
+
+	if criteria {
+		request.Criteria = mapCriteria(d)
+	}
+
+	_, _, err := conn.OnlineArchives.Update(context.Background(), projectID, clusterName, atlasID, &request)
+
+	if err != nil {
+		return fmt.Errorf("error updating Mongo Online Archive id: %s %s", atlasID, err.Error())
+	}
+
+	return resourceMongoDBAtlasOnlineArchiveRead(d, meta)
+}
+
+func fromOnlineArchiveToMap(in *matlas.OnlineArchive) map[string]interface{} {
+	// computed attribute
+	schemaVals := map[string]interface{}{
+		"cluster_name": in.ClusterName,
+		"atlas_id":     in.ID,
+		"paused":       in.Paused,
+		"state":        in.State,
+		"coll_name":    in.CollName,
+	}
+
+	// criteria
+	criteria := map[string]interface{}{
+		"type":              in.Criteria.Type,
+		"date_field":        in.Criteria.DateField,
+		"date_format":       in.Criteria.DateFormat,
+		"expire_after_days": int(in.Criteria.ExpireAfterDays),
+		// missing query check in client
+	}
+
+	// clean up criteria for empty values
+	for key, val := range criteria {
+		if isEmpty(val) {
+			delete(criteria, key)
+		}
+	}
+
+	schemaVals["criteria"] = criteria
+
+	// partitions fields
+	if len(in.PartitionFields) > 0 {
+		expected := make([]map[string]interface{}, 0, len(in.PartitionFields))
+		for _, partition := range in.PartitionFields {
+			if partition == nil {
+				continue
+			}
+
+			partition := map[string]interface{}{
+				"field_name": partition.FieldName,
+				"field_type": partition.FieldType,
+				"order":      partition.Order,
+			}
+
+			expected = append(expected, partition)
+		}
+		schemaVals["partition_fields"] = expected
+	}
+
+	return schemaVals
+}
+
+func fromOnlineArchiveToMapInCreate(in *matlas.OnlineArchive) map[string]interface{} {
+	localSchema := fromOnlineArchiveToMap(in)
+	criteria := localSchema["criteria"]
+	localSchema["criteria"] = []interface{}{criteria}
+
+	delete(localSchema, "partition_fields")
```

Comment:
Hello for this one, there is an issue because partition_fields is a list (it's order aware), For example the API returns a third element that is used in the date_field to query the collection and do the archive log. having a change of order sometimes from the API vs what you requested from the config leads to a config conflict.  So I preferred to honor the input from the user :)

Reply:
Thank you very much! : )